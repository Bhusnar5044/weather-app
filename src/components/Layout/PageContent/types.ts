import { PropsWithChildren } from 'react';

export interface Props extends PropsWithChildren {
    isLoading?: boolean;
}
