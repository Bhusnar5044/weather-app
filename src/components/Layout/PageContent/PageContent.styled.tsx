import { styled } from '@mui/material/styles';

export const PageContent = styled('main')`
    display: flex;
    flex-direction: column;
    gap: 2rem;
    overflow: auto;
    padding-top: 32px;
    width: 96%;
    margin: 0 auto;
    @media (min-width: 1536px) {
        max-width: 1536px;
    }
`;
