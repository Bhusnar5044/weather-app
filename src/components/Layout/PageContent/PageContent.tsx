import { FC, memo } from 'react';
import { Props } from './types';

export const PageContent: FC<Props> = memo(({ children }) => {
    return <main className="flex flex-col gap-2 w-full">{children}</main>;
});

PageContent.displayName = 'PageContent';
