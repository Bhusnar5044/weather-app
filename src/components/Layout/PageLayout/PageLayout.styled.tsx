import { styled } from '@mui/material';
import { theme } from '@theme';

export const PageLayoutStyled = styled('div')`
    display: grid;
    height: 100vh;
    width: 100vw;
    grid-template-areas:
        'header'
        'main';
    grid-template-columns: 1fr;
    grid-template-rows: 4.2em auto;
    background-color: ${theme.palette.grey[800]};

    & > div {
        padding: 0;
        margin: 2rem 0;
    }

    & > header {
        grid-area: header;
    }

    & > main {
        grid-area: main;
        transition: width 200ms;
        padding-bottom: 35px;
    }
`;
