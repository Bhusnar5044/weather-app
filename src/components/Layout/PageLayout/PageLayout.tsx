import { FC, PropsWithChildren, memo } from 'react';

export const PageLayout: FC<PropsWithChildren> = memo((props) => (
    <div className="bg-dark h-screen w-screen ">{props.children}</div>
));

PageLayout.displayName = 'PageLayout';
