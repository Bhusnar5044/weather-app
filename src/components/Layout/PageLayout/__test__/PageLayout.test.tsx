import { render, screen } from '@test-utils';
import PageContent from '../../PageContent';
import { PageLayout } from '../PageLayout';

describe('PageLayout component', () => {
    it('should render properly', () => {
        const { container } = render(
            <PageLayout>
                <PageContent>
                    <p>dummy Text</p>
                </PageContent>
            </PageLayout>,
        );
        expect(screen.getByText('dummy Text')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
