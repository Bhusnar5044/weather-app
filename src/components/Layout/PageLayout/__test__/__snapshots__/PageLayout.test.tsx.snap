// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout component should render properly 1`] = `
<div>
  <div
    class="bg-dark h-screen w-screen "
  >
    <main
      class="flex flex-col gap-2 w-full"
    >
      <p>
        dummy Text
      </p>
    </main>
  </div>
</div>
`;
